examples/progress-dashboard/app.py

```diff
@@ -0,0 +1,205 @@
+                ),
+                # x=0.5,
+                # y=-0.2,
+                showarrow=False,
+                xref="paper",
+                yref="paper",
+                font=dict(size=16),
+            )
+        ],
+    )
+
+    fig.add_annotation(
+        text=(
+            f"Current Progress: {progress['progress']:.1f}% complete<br>"
+            f"({progress['annotated']} out of {progress['total']} records annotated)"
+        ),
+        xref="paper",
+        yref="paper",
+        x=0.5,
+        y=1.1,
+        showarrow=False,
+        font=dict(size=18),
+        align="center",
+    )
+
+    return fig
+
+
+def create_treemap(user_annotations, total_records):
+    sorted_users = sorted(user_annotations.items(), key=lambda x: x[1], reverse=True)
+    color_scale = colors.qualitative.Pastel + colors.qualitative.Set3
+
+    labels, parents, values, text, user_colors = [], [], [], [], []
+
+    for i, (user, contribution) in enumerate(sorted_users):
+        percentage = (contribution / total_records) * 100
+        labels.append(user)
+        parents.append("Annotations")
+        values.append(contribution)
+        text.append(f"{contribution} annotations<br>{percentage:.2f}%")
+        user_colors.append(color_scale[i % len(color_scale)])
+
+    labels.append("Annotations")
+    parents.append("")
+    values.append(total_records)
+    text.append(f"Total: {total_records} annotations")
+    user_colors.append("#FFFFFF")
+
+    fig = go.Figure(
+        go.Treemap(
+            labels=labels,
+            parents=parents,
+            values=values,
+            text=text,
+            textinfo="label+text",
+            hoverinfo="label+text+value",
+            marker=dict(colors=user_colors, line=dict(width=2)),
+        )
+    )
+
+    fig.update_layout(
+        title_text="User contributions to the total end dataset",
+        height=500,
+        margin=dict(l=10, r=10, t=50, b=10),
+        paper_bgcolor="#F0F0F0",  # Light gray background
+        plot_bgcolor="#F0F0F0",  # Light gray background
+    )
+
+    return fig
+
+
+def get_datasets(client: rg.Argilla) -> List[rg.Dataset]:
+    return client.datasets.list()
+
+
+datasets = get_datasets(client)
+
+
+def update_dashboard(dataset_idx: int | None = None):
+    if dataset_idx is None:
+        return [None, None, None]
+
+    dataset = datasets[dataset_idx]
+    progress = get_progress(dataset)
+
+    gauge_chart = create_gauge_chart(progress)
+    treemap = create_treemap(progress["users"], progress["total"])
+
+    leaderboard_df = pd.DataFrame(
+        list(progress["users"].items()), columns=["User", "Annotations"]
+    )
+
+    leaderboard_df = leaderboard_df.sort_values(
+        "Annotations", ascending=False
+    ).reset_index(drop=True)
+
+    return gauge_chart, treemap, leaderboard_df
+
+
+with gr.Blocks() as demo:
+    gr.Markdown("# Argilla Dataset Dashboard")
+
+    datasets_dropdown = gr.Dropdown(label="Select your dataset")
```

Comment:
I think it would be nice to pre-select a dataset instead of entering on an empty screen.